.circleci/config.yml

```diff
@@ -148,6 +148,7 @@ jobs:
       - codecov/upload:
           flags: frontend
           file: coverage/coverage-final.json
+          when: always
```

Comment:
that's the [default](https://circleci.com/developer/orbs/orb/codecov/codecov#commands) but you wanted to make sure?

Reply:
Oops. I have to admit I hadn't tested - I took from the Trello discussion around codecov that reports weren't being uploaded when tests failed, and hence the flip flopping in coverage? I clearly misunderstood. Can you clarify when you have a second what you think the issue has been? Thanks 🙏

Reply:
@jamescrowley I suspect the following issue: e.g. when running the BE tests with pytest, the python-coverage package will not generate any report if tests fail but I have to double check this.

If no report has been generated, codecov has no data for upload, and then coverage numbers will drop.

Reply:
at least on GitHub I believe codecov's default behaviour is to fail if there is no report - [handle_no_reports_found](https://github.com/codecov/codecov-action). I can't see any reference to what the behaviour is on circleci.

I was hoping my other config change for requiring 2 reports - `after_n_builds: 2` - may tackle this? Do you happen to remember a PR where you've seen this happen recently?

Reply:
I've removed the attributes from the PR given they do nothing!

Reply:
in #1446 test-front fails and [no reports are uploaded to codecov](https://app.circleci.com/pipelines/github/boxwise/boxtribute/5137/workflows/5021dcd5-ac3a-43d7-a44e-0ae88a726d60/jobs/28042)